gradio/image_utils.py

```diff
@@ -27,7 +28,7 @@ def format_image(
         return np.array(im)
     elif type == "filepath":
         path = processing_utils.save_pil_to_cache(
-            im, cache_dir=cache_dir, name=name, format=fmt or "png"  # type: ignore
+            im, cache_dir=cache_dir, name=name, format=fmt or format  # type: ignore
```

Comment:
The only risk (admittedly small) I can see here is that there are some image formats that PIL can read but not save as: https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#read-only-formats

They all look very rare to me, but perhaps a more robust approach would be to try / catch this, and save as "png" as a fallback

Reply:
Yea that's a good point